test/validation/api/ipsec/ipsec.h

```diff
@@ -30,6 +30,9 @@ int ipsec_in_inline_init(void);
 int ipsec_out_inline_init(void);
 
 int ipsec_suite_init(void);
+int ipsec_suite_plain_init(void);
+int ipsec_suite_sched_init(void);
```

Comment:
Now with the new suite init functions, maybe ipsec_suite_init() could be renamed something more specific (ipsec_suite_sync_init?).


The suite init functions do not seem to reinitialize suite_context to default values. It seems to me that e.g. suite_context.queue could be left to a stale handle if sync test suite is run after an async or inline suite. Currently it does not happen because those suites are run in different processes, but for robustness I think it would be better
not to rely on that but have suite_init initialize everything in suite context.

Reply:
ok, I see. It is a bit messy since there are both global (with the lifetime of the process) and suite specific stuff in suite_context. But it is not the fault of this patch, so this is fine.

Reply:
Ack.